### PR TITLE
[ACM-4770]: Porting external DNS docs from upstream to ACM docs

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-4770

This PR removes a link to the upstream docs and ports over the relevant info about [external DNS from upstream](https://hypershift-docs.netlify.app/how-to/aws/external-dns/) to the hosted control planes docs.

### CODE REVIEW

**Comment by lahinson** [comment]
Hi @rokej - When you can, please take a look at this PR, which ports the external DNS docs from the upstream HyperShift docs to the official docs in ACM. Thank you!

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 1]
```diff
@@ -109,11 +109,50 @@ aws route53 create-hosted-zone --name $BASE_DOMAIN --caller-reference $(whoami)-
...
 == Enabling external DNS
 
+Because the control plane and the data plane are separate in hosted control planes, you can configure DNS in two independent areas:
+
+* Ingress for workloads within the hosted cluster, such as `*.apps.service-consumer-domain.com`
+* Ingress for service endpoints within the management cluster, such as API or OAUTH endpoints through `*.service-provider-domain.com`
+
+The input for the `hostedCluster.spec.dns` dictates the Ingress for workloads within the hosted cluster. The input for `hostedCluster.spec.services.routePublishingStrategy.hostname` dictates the Ingress for service endpoints within the management cluster.
+
+External DNS is useful for setups with public endpoints, such as `Public` or `PublicandPrivate`. For a private setup, all endpoints must be accessible through `.hypershift.local`, which contains the CNAME records to the private link endpoint services.
```
Comment (rokej) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc]: PublicandPrivate  --> PublicAndPrivate

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 2]
```diff
@@ -109,11 +109,50 @@ aws route53 create-hosted-zone --name $BASE_DOMAIN --caller-reference $(whoami)-
 [#hosted-enable-ext-dns-aws]
 == Enabling external DNS
 
+Because the control plane and the data plane are separate in hosted control planes, you can configure DNS in two independent areas:
+
+* Ingress for workloads within the hosted cluster, such as `*.apps.service-consumer-domain.com`
+* Ingress for service endpoints within the management cluster, such as API or OAUTH endpoints through `*.service-provider-domain.com`
+
+The input for the `hostedCluster.spec.dns` dictates the Ingress for workloads within the hosted cluster. The input for `hostedCluster.spec.services.routePublishingStrategy.hostname` dictates the Ingress for service endpoints within the management cluster.
```
Comment (rokej) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc]: I think `hostedCluster.spec.services.routePublishingStrategy.hostname` should be `hostedCluster.spec.services.servicePublishingStrategy.route.hostname`.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 3]
```diff
@@ -128,29 +167,171 @@ If you plan to provision hosted control plane clusters with service-level DNS (e
...
+*Note:* Disaster recovery backup for the secret is not automatically enabled. To add the label that enables the `hypershift-operator-external-dns-credentials` secret to be backed up for disaster recovery, enter the following command:
 +
 ----
 oc label secret hypershift-operator-external-dns-credentials -n local-cluster cluster.open-cluster-management.io/backup=""
 ----
 
+[#create-public-dns-hosted-zone-aws]
+=== Creating the public DNS hosted zone
+
+You can create the public DNS hosted zone in the AWS Route 53 management console:
```
Comment (rokej) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc]: You can create the public DNS hosted zone to be used as the external DNS domain-filter  in the AWS Route 53 management console:

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 4]
```diff
@@ -128,29 +167,171 @@ If you plan to provision hosted control plane clusters with service-level DNS (e
...
+
+
+[#deploy-cluster-cli-external-dns-aws]
+=== Deploying a cluster by using the command line interface and external DNS
+
+To deploy the `hypershift` and `external-dns` Operators when the external public hosted zone already exists, take the following steps:
+
+. Ensure that the public hosted zone exists. In the following example, the zone is `service-provider-domain.com`.
+
+. Enter the following commands:
```
Comment (rokej) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc]: Remove this step and the `hypershift install` command sample

**Comment by rokej** [comment]
@enxebre could you please review as well? Thanks.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 1 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: Good catch! Fixing now.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 2 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: Fixed

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 4 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: done

**Comment by lahinson** [comment]
Thanks so much for your review, @rokej! I incorporated your feedback.

**Review by rokej — APPROVED** [review]
Thank you

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 5]
```diff
@@ -109,11 +109,50 @@ aws route53 create-hosted-zone --name $BASE_DOMAIN --caller-reference $(whoami)-
...
 == Enabling external DNS
 
+Because the control plane and the data plane are separate in hosted control planes, you can configure DNS in two independent areas:
+
+* Ingress for workloads within the hosted cluster, such as `*.apps.service-consumer-domain.com`
+* Ingress for service endpoints within the management cluster, such as API or OAUTH endpoints through `*.service-provider-domain.com`
+
+The input for the `hostedCluster.spec.dns` dictates the Ingress for workloads within the hosted cluster. The input for `hostedCluster.spec.services.servicePublishingStrategy.route.hostname` dictates the Ingress for service endpoints within the management cluster.
+
+External DNS is useful for setups with public endpoints, such as `Public` or `PublicAndPrivate`. For a private setup, all endpoints must be accessible through `.hypershift.local`, which contains the CNAME records to the private link endpoint services.
```
Comment (enxebre) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc]: this changed a bit recently, even for private we create the public dns record in AWS cc @sjenning to corroborate

Comment (sjenning) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: Yes, it is true that for `Private` clusters, we do create a DNS record for `Services` that  have a publishing type of `Route` and a `route.hostname` set.  In the case of `Private`, the DNS record points to a private IP of a VPC Endpoint in the customer VPC.

Comment (sjenning) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: This is the PR that did that btw https://github.com/openshift/hypershift/pull/2286

Public DNS registration of private endpoints is only supported on the `APIServer` and `OAuth` services.

Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: Thanks @sjenning! Is this revision of line 119 accurate?

>External DNS is useful for setups with public endpoints, such as `Public` or `PublicAndPrivate`. For a private setup, the DNS record points to a private IP of a Virtual Private Cloud (VPC) endpoint in your VPC. Public DNS registration of private endpoints is supported only on the `APIServer` and `OAuth` services.

Comment (sjenning) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: How about this?
> External DNS creates name records for hosted cluster `Services` that specify a publishing type of `LoadBalanacer` or `Route` and provide a `hostname` for that publishing type. For hosted clusters with `Private` or `PublicAndPrivate` endpoint access types, only the `APIServer` and `OAuth` services support hostnames.  For `Private` hosted clusters, the DNS record resolves to a private IP of a Virtual Private Cloud (VPC) Endpoint in your VPC.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 6]
```diff
@@ -109,11 +109,50 @@ aws route53 create-hosted-zone --name $BASE_DOMAIN --caller-reference $(whoami)-
...
+* `Ignition`
+
+Each of those services is exposed by using `servicePublishingStrategy` in the `HostedCluster` specification. By default, for the `LoadBalancer` and `Route` types of `servicePublishingStrategy`, you publish the service in one of two ways: 
+
+* By using the hostname of the load balancer that is in the status of the `Service` with the `LoadBalancer` type
+* In the `status.host` field of the `Route`
+
+However, when you deploy hosted control planes in a managed service context, those methods can expose the Ingress subdomain of the underlying management cluster and limit options for the management cluster lifecycle and disaster recovery.
+
+When a DNS indirection layer is on top of the `LoadBalancer` and `Route` publishing types, a managed service operator can publish all public hosted cluster services by using a service-level domain. This architecture allows remapping on the DNS name to a new `LoadBalancer` or `Route` and does not expose the Ingress domain of the management cluster. Hosted control planes uses external DNS to achieve that indirection layer.
```
Comment (swopebe) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc]: Can we say "is layered on the load balancer" here to eliminate "top"?

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 7]
```diff
@@ -128,29 +167,144 @@ If you plan to provision hosted control plane clusters with service-level DNS (e
...
+        hostname: oauth-example.service-provider-domain.com
+      type: Route
+  - service: Konnectivity
+    servicePublishingStrategy:
+      type: Route
+  - service: Ignition
+    servicePublishingStrategy:
+      type: Route
+----
+
```
Comment (swopebe) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc:274]: There is something in the code graying out the content here.

Comment (swopebe) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc:274]: May be the ... in the YAML, which we don't have much in our doc, I don't think.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 8]
```diff
@@ -128,29 +167,144 @@ If you plan to provision hosted control plane clusters with service-level DNS (e
...
+    --pull-secret ${HOME}/pull_secret.json \
+    --release-image quay.io/openshift-release-dev/ocp-release:4.12.0-ec.3-x86_64 \
+    --external-dns-domain=service-provider-domain.com \ <2>
+    --endpoint-access=PublicAndPrivate <3>
+----
+
+<1> Points to the public hosted zone, `service-consumer-domain.com`, which is typically in an AWS account that the service consumer owns.
+<2> Points to the public external DNS hosted zone, `service-provider-domain.com`, which is typically in an AWS account that the service provider owns.
+<3> Set as `PublicAndPrivate.` The external DNS can be used with only `Public` or `PublicAndPrivate` configurations.
+
```
Comment (swopebe) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc:334]: This whole section is doing the same thing.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 9]
```diff
@@ -257,12 +411,12 @@ NAME               AVAILABLE   DEGRADED   PROGRESSING
...
 
-Your HyperShift add-on is installed and the hosting cluster is available to create and manage HyperShift clusters.
+Your HyperShift add-on is installed and the hosting cluster is available to create and manage hosted clusters.
 
 [#hosted-install-cli]
-== Installing the hosted control planes CLI
+== Installing the hosted control planes command line interface
 
-The hosted control planes (HyperShift) CLI is used to create and manage {ocp-short} hosted control plane clusters. After enabling the hosted control planes feature, you can install the hosted control planes CLI by completing the following steps:
+The hosted control planes (`hypershift`) command line interface is used to create and manage {ocp-short} hosted control plane clusters. After you enable the hosted control planes feature, you can install the hosted control planes command line interface by completing the following steps:
```
Comment (swopebe) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc:450]: Same here

File: clusters/hosted_control_planes/configure_hosted_aws.adoc [hunk 10]
```diff
@@ -299,6 +453,4 @@ hypershift create cluster aws --help
 
 * For more information about the AWS credential secret, see link:https://hypershift-docs.netlify.app/how-to/aws/deploy-aws-private-clusters/[Deploying AWS private clusters] in the HyperShift documentation.
```
Comment (swopebe) [inline clusters/hosted_control_planes/configure_hosted_aws.adoc]: can we build this section like: https://github.com/stolostron/rhacm-docs/blob/2.8_stage/business_continuity/volsync/volsync_replicate.adoc#additional-resources

**Review by swopebe — CHANGES_REQUESTED** [review]

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 6 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: Sure! I'll update that wording.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 7 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc:274]: I'll take a look

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 10 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc]: Sure. So you mean that you want one main lead-in sentence and then the list of resources?

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 7 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc:274]: hmm...removing the ellipsis doesn't seem to make a difference. I'll keep playing with it.

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 9 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc:450]: fixed!

File: clusters/hosted_control_planes/configure_hosted_aws.adoc (hunk 7 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_aws.adoc:274]: I figured it out! I had to remove the + from lines 220 and 222.

**Comment by lahinson** [comment]
@swopebe Thanks so much for your review! I think I've incorporated all of your feedback. Let me know what you think when you can.

**Comment by lahinson** [comment]
@swopebe When you can, PTAL and let me know if the latest changes I made to this PR look okay. I appreciate it!

**Review by swopebe — APPROVED** [review]
thanks for pulling me in!

**Pull request merged**